### ISSUE TITLE
OpenJDK 10 - compatibility issue

### ISSUE THREAD

**Issue by rjtokenring** [OpenHFT/Chronicle-Map#155]
Hi, taking a simple working code with sun jdk 1.8 and compiling/running with OpenJDK10, errors are thrown. 
Here is stack trace:

`net.openhft.chronicle.core.Jvm: Unable to determine max direct memory
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by net.openhft.chronicle.core.Jvm (file:/xxxxxxxx/net/openhft/chronicle-core/1.15.1/chronicle-core-1.15.1.jar) to field java.nio.Bits.reservedMemory
WARNING: Please consider reporting this to the maintainers of net.openhft.chronicle.core.Jvm
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
Exception in thread "main" java.lang.IllegalStateException: Key size in serialized form must be configured in ChronicleMap, at least approximately.
Use builder.averageKey()/.constantKeySizeBySample()/.averageKeySize() methods to configure the size
	at net.openhft.chronicle.map.ChronicleMapBuilder.preMapConstruction(ChronicleMapBuilder.java:1865)
	at net.openhft.chronicle.map.ChronicleMapBuilder.preMapConstruction(ChronicleMapBuilder.java:1848)
	at net.openhft.chronicle.map.ChronicleMapBuilder.newMap(ChronicleMapBuilder.java:1832)
	at net.openhft.chronicle.map.ChronicleMapBuilder.lambda$createWithFile$3(ChronicleMapBuilder.java:1633)
	at net.openhft.chronicle.map.ChronicleMapBuilder.lambda$fileLockedIO$1(ChronicleMapBuilder.java:257)
	at java.base/java.util.concurrent.ConcurrentHashMap.compute(ConcurrentHashMap.java:1922)
	at net.openhft.chronicle.map.ChronicleMapBuilder.fileLockedIO(ChronicleMapBuilder.java:254)
	at net.openhft.chronicle.map.ChronicleMapBuilder.createWithFile(ChronicleMapBuilder.java:1631)
	at net.openhft.chronicle.map.ChronicleMapBuilder.createPersistedTo(ChronicleMapBuilder.java:1549)
	at net.openhft.chronicle.map.ChronicleMapBuilder.createOrRecoverPersistedTo(ChronicleMapBuilder.java:1571)
	at net.openhft.chronicle.map.ChronicleMapBuilder.createOrRecoverPersistedTo(ChronicleMapBuilder.java:1560)
	at net.openhft.chronicle.map.ChronicleMapBuilder.createOrRecoverPersistedTo(ChronicleMapBuilder.java:1554)`

Code is 

`ChronicleMap<LongValue, int[]> map = null;
		try {
			map = ChronicleMapBuilder
					.of(LongValue.class, int[].class)
					.name("map")
					.entries(HOW_MANY)
					.averageValue(new int[VALUE_LENGTH])
					.createOrRecoverPersistedTo(path);`

**Comment by epickrram** [comment]
Are you sure it works on Java 1.8? It looks as though this is the issue:

```
java.lang.IllegalStateException: Key size in serialized form must be configured in ChronicleMap, at least approximately. Use builder.averageKey()/.constantKeySizeBySample()/.averageKeySize() methods to configure the size
```

The warnings about reflective access should not cause the program to terminate.

**Comment by dpisklov** [comment]
@epickrram hi Mark :)

**Comment by MartyIX** [comment]
`net.openhft.chronicle.core.Jvm: Unable to determine max direct memory` - This warning is kind of annoying.

**Comment by rjtokenring** [comment]
This is a working code. It work perfectly with jdk 1.8.

[TestMap.zip](https://github.com/OpenHFT/Chronicle-Map/files/2204947/TestMap.zip)

Yes, warning on reflect are annoying but does not create runtime issues.

**Comment by dpisklov** [comment]
Chronicle at the moment only supports Java 8. The next version we plan to support is Java 11, although that support is not quite there yet. Any intermediate versions between those will be supported for paid support customers only.

**Comment by titanium9** [comment]
Hello,
What is the status of chronicle-map and Java 11 support?
Do you know when it will be available?
Thanks,

**Comment by dpisklov** [comment]
Chronicle supports Java 11 since chronicle-bom-2.17.0

**Comment by hft-team-city** [comment]
Released in Chronicle-Map-3.17.0 (https://github.com/OpenHFT/Chronicle-Map/releases/chronicle-map-3.17.0), BOM-2.17.48 (https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.17.48)

**Comment by hft-team-city** [comment]
Released in [Chronicle-Map-3.17.0](https://github.com/OpenHFT/Chronicle-Map/releases/chronicle-map-3.17.0), [BOM-2.17.50](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.17.50)

**Comment by hft-team-city** [comment]
Released in [Chronicle-Map-3.17.0](https://github.com/OpenHFT/Chronicle-Map/releases/chronicle-map-3.17.0), [BOM-2.17.52](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.17.52)

**Comment by hft-team-city** [comment]
Released in [Chronicle-Map-3.17.0](https://github.com/OpenHFT/Chronicle-Map/releases/chronicle-map-3.17.0), [BOM-2.17.53](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.17.53)

**Comment by hft-team-city** [comment]
Released in [Chronicle-Map-3.17.0](https://github.com/OpenHFT/Chronicle-Map/releases/chronicle-map-3.17.0), [BOM-2.17.60](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.17.60)

**Comment by hft-team-city** [comment]
Released in [Chronicle-Map-3.17.0](https://github.com/OpenHFT/Chronicle-Map/releases/chronicle-map-3.17.0), [BOM-2.17.72](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.17.72)

**Comment by hft-team-city** [comment]
Released in [Chronicle-Map-3.17.0](https://github.com/OpenHFT/Chronicle-Map/releases/chronicle-map-3.17.0), [BOM-2.17.74](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.17.74)
